src/Orleans/Configuration/ClientConfiguration.cs

```diff
@@ -74,6 +74,13 @@ public enum GatewayProviderType
         public string DataConnectionString { get; set; }
 
         /// <summary>
+        /// the id of the cluster to which this client is connecting. 
+        /// This argument is optional, but is needed in order for the client to receive
+        /// observer messages from different clusters.
+        /// </summary>
+        public string ClusterId { get; set; }
```

Comment:
Doing it automatically seems possible in principle but there are some questions and potential risks with it: (1) the clientId is used in the code before the client connects, so we don't know the cluster id yet unless it is configured. I could change the clientId once connected but that is not so nice, e.g. log entries are inconsistent, other code that reads it may get confused..  (2) I think clients do sometimes reconnect to a different silo. So we have to ensure the other silo is in the same cluster.

I went for the configuration option because it is very simple and low risk form our perspective. I agree it makes life slightly more complicated for our users since they have to configure the client with the cluster id.